flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV100.java

```diff
@@ -335,4 +338,18 @@ public CatalogColumnStatisticsDataDate toFlinkDateColStats(ColumnStatisticsData
 			throw new CatalogException("Failed to get table schema from deserializer", e);
 		}
 	}
+
+	@Override
+	public Set<String> listBuiltInFunctions() {
+		// FunctionInfo doesn't have isBuiltIn() API to tell whether it's a builtin function or not
+		// prior to Hive 1.2.0
+		throw new UnsupportedOperationException("Listing built in functions are not supported until Hive 1.2.0");
+	}
+
+	@Override
+	public Optional<FunctionInfo> getBuiltInFunctionInfo(String name) {
+		// FunctionInfo doesn't have isBuiltIn() API to tell whether it's a builtin function or not
+		// prior to Hive 1.2.0
+		throw new UnsupportedOperationException("Getting built in functions are not supported until Hive 1.2.0");
+	}
```

Comment:
I think we have two options: 1. Document the restriction (with a JIRA to track) 2. Just list all functions which might include non-built-in ones.

Reply:
I haven't seen a requirement to list/get non built-in functions yet, thus probably better not to do it now. Limitations will be mentioned when I write docs for HiveModule.